compiler/quilt/tools/command.py

```diff
@@ -934,28 +942,41 @@ def install(package, hash=None, version=None, tag=None, force=False):
 
     print("Downloading package metadata...")
 
-    if version is not None:
-        response = session.get(
-            "{url}/api/version/{owner}/{pkg}/{version}".format(
-                url=get_registry_url(team),
-                owner=owner,
-                pkg=pkg,
-                version=version
+    try:
+        if version is not None:
+            response = session.get(
+                "{url}/api/version/{owner}/{pkg}/{version}".format(
+                    url=get_registry_url(team),
+                    owner=owner,
+                    pkg=pkg,
+                    version=version
+                )
             )
-        )
-        pkghash = response.json()['hash']
-    elif tag is not None:
-        response = session.get(
-            "{url}/api/tag/{owner}/{pkg}/{tag}".format(
-                url=get_registry_url(team),
-                owner=owner,
-                pkg=pkg,
-                tag=tag
+            pkghash = response.json()['hash']
+        elif tag is not None:
+            response = session.get(
+                "{url}/api/tag/{owner}/{pkg}/{tag}".format(
+                    url=get_registry_url(team),
+                    owner=owner,
+                    pkg=pkg,
+                    tag=tag
+                )
             )
-        )
-        pkghash = response.json()['hash']
-    else:
-        pkghash = _match_hash(session, team, owner, pkg, hash)
+            pkghash = response.json()['hash']
+        else:
+            pkghash = _match_hash(session, team, owner, pkg, hash)
+    except CommandException as e:
+        logged_in_team = _find_logged_in_team()
+        if team is None and logged_in_team is not None:
+            notfoundstr = ("Package {owner}/{pkg} does not exist " +
```

Comment:
This is really fragile. At most, we should check for a 404... But I guess that requires changing or disabling the `_handle_response` function.